nixos/modules/misc/version.nix

```diff
@@ -60,6 +61,35 @@ in
       description = "The NixOS version suffix (e.g. <literal>1160.f2d4ee1</literal>).";
     };
 
+    nixosState = rec {
+      isSupported = mkOption {
+        readOnly = true;
+        internal = true;
+        type = types.bool;
+        default = nixosState.state == "supported";
+        description = "Whether or not this version of NixOS is supported.";
+      };
+
+      state = mkOption {
+        readOnly = true;
+        internal = true;
+        type = types.nullOr (types.enum ["supported" "deprecated" "unsupported"]);
+        default = nixosState.state;
+        description = "Support state of this version of NixOS. Both depreciated and unsupported are considered to be unsupported.";
```

Comment:
I think `deprecated` is supported, but perhaps you mean "both deprecated and unsupported will raise a warning."

Reply:
deprec**i**ated → deprecated.

<sup>Unless this is because the older releases diminish in value over a period of time?</sup>